prow/githuboauth/githuboauth.go

```diff
@@ -109,7 +128,16 @@ func (ga *Agent) HandleLogin(client OAuthClient) http.HandlerFunc {
 			return
 		}
 
-		redirectURL := client.AuthCodeURL(state, oauth2.ApprovalForce, oauth2.AccessTypeOnline)
+		newClient := &RealOAuthClient{
+			&oauth2.Config{
+				ClientID:     client.GetConfig().ClientID,
+				ClientSecret: client.GetConfig().ClientSecret,
+				RedirectURL:  client.GetConfig().RedirectURL + "?dest=" + destPage + "&rerun_status=" + rerunStatus,
```

Comment:
Will this work? Doesn't it need to 100% match what is registered with GitHub for the OAuth flow to succeed?

Reply:
It works. It just needs to include the expected URL, which in this case is `client.GetConfig().RedirectURL`

Reply:
Awesome